Modules/_randommodule.c

```diff
@@ -510,6 +510,44 @@ _random_Random_getrandbits_impl(RandomObject *self, int k)
     return result;
 }
 
+/*[clinic input]
+
+_random.Random.getrandbytes
+
+  self: self(type="RandomObject *")
+  n: Py_ssize_t
+  /
+
+Generate n random bytes.
+[clinic start generated code]*/
+
+static PyObject *
+_random_Random_getrandbytes_impl(RandomObject *self, Py_ssize_t n)
+/*[clinic end generated code: output=adaef69179dd1a56 input=95028b66d39d1bcc]*/
+{
+    if (n < 1) {
+        PyErr_SetString(PyExc_ValueError,
+                        "number of bytes must be greater than zero");
+        return NULL;
+    }
+
+    PyObject *bytes = PyBytes_FromStringAndSize(NULL, n);
+    if (bytes == NULL) {
+        return NULL;
+    }
+    uint8_t *ptr = (uint8_t *)PyBytes_AS_STRING(bytes);
+
+    do {
+        uint32_t word = genrand_uint32(self);
+        size_t chunk = Py_MIN(n, 4);
+        memcpy(ptr, &word, chunk);
+        ptr += chunk;
+        n -= chunk;
```

Comment:
I would write
```suggestion
        if (n < 4) {
            memcpy(ptr, &word, n);
            break;
        }
        memcpy(ptr, &word, 4);
        ptr += 4;
        n -= 4;
```
`memcpy()` and incrementing/decrementing with constant size may be faster.

Also do not forget about endianess.

Reply:
> I would write (...)

Done.

> Also do not forget about endianess.

I don't know how to handle the endianness. Should use htonl() to convert word to big endian?

Reply:
Yes, you can use `htonl()` or swap bytes manually, as it is done in many places in CPython for performace. Search WORDS_BIGENDIAN, PY_BIG_ENDIAN and PY_LITTLE_ENDIAN. Perhaps the most efficient way is:
```c
x = ...
x = ((x & 0x00FF00FFu) << 8) | ((x >> 8) & 0x00FF00FFu);
x = ((x & 0x0000FFFFu) << 16) | ((x >> 16) & 0x0000FFFFu));
```

Reply:
I created https://bugs.python.org/issue40302 "Add _Py_bswap32() function to pyport.h".